### PR TITLE
Gracefully handle pending changes for contributions

### LINKED ISSUE #120: Solution for visualizing pending data
There are many options for this, and it needs to be solved on multiple levels.

Basic requirements:

1. Keep/read pending status and tx info the same way for all models/records
2. Disable UI interactions on pending objects

### PR DESCRIPTION
**Pull request by skddc**
This adds the pending tx data to pending contributions (after adding or vetoing, until the tx is mined). It also disables the veto button while pending.

WIP until I've added the same for contributor additions/changes. Also, it's branched out from #156, <del>which hasn't been merged yet.</del>

closes #120

### CODE REVIEW

**Comment by skddc** [comment]
Looking for reviews now. Doing contributors in a separate PR.

File: app/controllers/dashboard/contributors/show.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
   roleName: computed('model.isCore', 'totalKreditsEarned', function() {
     if (this.model.isCore) return 'Core Contributor';
     if (this.model.totalKreditsEarned <= 5000) return 'Newcomer';
     return 'Contributor';
-  })
+  }),
+
+  ipfsGatewayUrl: computed(function() {
+    return config.ipfs.gatewayUrl;
+  }).volatile()
```
Comment (galfert) [inline app/controllers/dashboard/contributors/show.js]: Why does this property need to be volatile?

**Comment by skddc** [comment]
@galfert Does your comment mean that you took this for a spin?

File: app/controllers/dashboard/contributors/show.js (hunk 1 above)
Comment (galfert) [reply app/controllers/dashboard/contributors/show.js]: No, marking a property as volatile just deactivates caching and makes it recalculate on every access. It also prevents property notifications.

The `volatile()` modifier [got deprecated recently](https://deprecations.emberjs.com/v3.x/#toc_computed-property-volatile). So I think just using an empty string as the dependency would be better here.

**Comment by galfert** [comment]
I only looked at the code. Besides my one comment it looked ok to me.

**Comment by skddc** [comment]
Could you please test it?

**Comment by skddc** [comment]
Note to whoever didn't use this feature before: in order to get longer, more realistic confirmation times from your dev chain, you can add `block-time` to the `devchain` npm script, like so:

```diff
-    "devchain": "aragon devchain --port 7545",
+    "devchain": "aragon devchain --port 7545 --block-time 20",
```

**Comment by galfert** [comment]
I tested the PR and confirmed everything to work.

I fixed the merge conflict and changed the `confirmation-status` helper to detect the pending changes. I also removed the `volatile()` property modifier everywhere, as it was already throwing a deprecation warning.

**Pull request merged**